components/function-controller/internal/webhook/conversion_webhook_test.go

```diff
@@ -0,0 +1,508 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	serverlessv1alpha1 "github.com/kyma-project/kyma/components/function-controller/pkg/apis/serverless/v1alpha1"
+	serverlessv1alpha2 "github.com/kyma-project/kyma/components/function-controller/pkg/apis/serverless/v1alpha2"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testRepoName = "test-repo"
+)
+
+func TestConvertingWebhook_convertFunction(t *testing.T) {
+	testGitRepo := &serverlessv1alpha1.GitRepository{
+		ObjectMeta: metav1.ObjectMeta{Name: testRepoName, Namespace: "test"},
+
+		Spec: serverlessv1alpha1.GitRepositorySpec{
+			URL: "https://github.com/kyma-project/kyma.git",
+			Auth: &serverlessv1alpha1.RepositoryAuth{
+				Type:       serverlessv1alpha1.RepositoryAuthBasic,
+				SecretName: "secret_name",
+			},
+		},
+	}
+
+	scheme := runtime.NewScheme()
+	_ = serverlessv1alpha1.AddToScheme(scheme)
+	_ = serverlessv1alpha2.AddToScheme(scheme)
+
+	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(testGitRepo).Build()
+	w := NewConvertingWebhook(client, scheme)
+	tests := []struct {
+		name          string
+		src           runtime.Object
+		wantDst       runtime.Object
+		wantVersion   string
+		wantErr       bool
+		checkRecreate bool
+	}{
+		{
+			name: "v1alpha1 to v1alpha2 inline function",
```

Comment:
I would add test case when resources are not filled or even consider to remove all resources in conversion scenarios to make them smaller and add special test case with resources.